galsim/chromatic.py

```diff
@@ -107,34 +106,28 @@ class ChromaticObject(object):
     # attribute in the case of compound classes like `ChromaticSum` and `ChromaticConvolution`.
 
     def __init__(self, obj):
-        if not isinstance(obj, (galsim.GSObject, ChromaticObject)):
-            raise TypeError("Can only directly instantiate ChromaticObject with a GSObject "+
-                            "or ChromaticObject argument.")
-        # TODO: Once we convert to a fully immutable style (when the mutating methods are
-        #       eventually removed), we can get rid of this copy() call.  Probably lots of others
-        #       as well...
-        self.obj = obj.copy()
+        self.obj = obj
         if isinstance(obj, galsim.GSObject):
             self.separable = True
-            self.SED = lambda w: 1.0
+            self.SED = galsim.SED('1')  # uniform flambda = 1
             self.wave_list = np.array([], dtype=float)
-        elif obj.separable:
-            self.separable = True
-            self.SED = obj.SED
+        elif isinstance(obj, ChromaticObject):
+            self.separable = obj.separable
             self.wave_list = obj.wave_list
+            if self.separable: self.SED = obj.SED
         else:
-            self.separable = False
-            self.SED = lambda w: 1.0
-            self.wave_list = np.array([], dtype=float)
-        # Some private attributes to handle affine transformations
-        # _A is a 3x3 augmented affine transformation matrix that holds both translation and
-        # shear/rotate/dilate specifications.
-        # (see http://en.wikipedia.org/wiki/Affine_transformation#Augmented_matrix)
-        self._A = lambda w: np.matrix(np.identity(3), dtype=float)
-        # _fluxFactor holds a wavelength-dependent flux rescaling.  This is only needed because
-        # a wavelength-dependent dilate(f(w)) is implemented as a combination of a
-        # wavelength-dependent expansion and wavelength-dependent flux rescaling.
-        self._fluxFactor = lambda w: 1.0
+            raise TypeError("Can only directly instantiate ChromaticObject with a GSObject "+
+                            "or ChromaticObject argument.")
+
+    def __repr__(self):
+        return 'galsim.ChromaticObject(%r)'%self.obj
+
+    def __str__(self):
+        return 'galsim.ChromaticObject(%s)'%self.obj
+
+    def __eq__(self, other): return repr(self) == repr(other)
+    def __ne__(self, other): return not self.__eq__(other)
+    def __hash__(self): return hash(repr(self))
 
     def setupInterpolation(self, waves, oversample_fac=1.):
```

Comment:
I'm wondering about the name of this method in the refactored code.  I think `setupInterpolation` sounds more like you're manipulating the current object (which used to actually be the case) instead of returning a new one.  Maybe just `interpolate` to go along with `dilate`, `shear`, etc.?